File: Issue/load.java
import java.io.*;
import java.util.*;

public class LoadExtraction {
	final static String[] locationMark = {"l:in", "l:at", "l:on", "l:from"};
	final static String[] timeMark = {"t:in", "t:at", "t:on", "t:from", "t:during", "t:between",
		"t:before", "t:after", "t:while", "t:when"};

	public static void main(String[] args) throws IOException{
		//Read output of extraction
		FileInputStream fstream = new FileInputStream(args[0]);
		DataInputStream input = new DataInputStream(fstream);
		BufferedReader reader = new BufferedReader(new InputStreamReader(input));
		ArrayList<String> outputLines= new ArrayList<String>();
		String readLine;
		while ((readLine = reader.readLine()) != null){
			outputLines.add(readLine);
		}
		input.close();
		
		//Process output of extraction
		ArrayList<String> loadLines = new ArrayList<String>();
		int extractionCandidate = 0;
		ArrayList<String> sentenceExtraction = new ArrayList<String>();
		for (String line: outputLines){
			if (line.equals("")){
				if (extractionCandidate == 1){
				  	for (String i: sentenceExtraction){
				  		loadLines.add(i);
					}
				}
				extractionCandidate = 0;
				sentenceExtraction.clear();
			}
			else{
				if (Character.isDigit(line.charAt(0))){
					extractionCandidate = 1;
					String content = "";
					if (line.charAt(5) != '('){
						int markPoint = line.indexOf('(');
						String keyWord = line.substring(5,markPoint);
						int startPoint = line.indexOf(':') + 2;
						content = '(' + keyWord + ')' + line.substring(startPoint, line.length() - 1);
					}
					else
						content = line.substring(6, line.length() - 1);
					String[] preLoadLine = content.split("; ");
					ArrayList<String> loadLine = new ArrayList<String>();
					loadLine.addAll(Arrays.asList(preLoadLine));
					if (loadLine.size() >= 3){
						loadLine = processTuple(loadLine);
						sentenceExtraction.addAll(loadLine);
					}			  					
				}
			}
		}
		PrintWriter outputWriter = new PrintWriter("extraction.txt");
		for (String i: loadLines){
			outputWriter.print(i);
			outputWriter.print('\n');
		}
	    outputWriter.close();	    
	    System.out.print("Finished");
	}
	
	private static String processTimeLocation(ArrayList<String> conditionList, char Mark, String conditionBuffer){
		if (conditionList.size() ==0)
			return conditionBuffer;
		else{
			String condition = conditionList.get(0);
			conditionList.remove(0);
			char[] conditionChar = condition.toCharArray();
			String[] conditionWord = condition.split(" ");
			if (conditionChar[0] == 'L' && conditionChar[1] == ':'){
				if (Collections.frequency(Arrays.asList(locationMark), conditionWord[0].toLowerCase()) == 0){
					char[] prepositionChar = Arrays.copyOfRange(conditionChar, 2, conditionWord[0].length());
					String prepositionWord = new String(prepositionChar);
					List<String> joined = new ArrayList<String>();
					joined.add(prepositionWord);
					joined.addAll(Arrays.asList(Arrays.copyOfRange(conditionWord, 1, conditionWord.length)));
					conditionBuffer += " in " + String.join(" ", joined);
					return processTimeLocation(conditionList, 'L', conditionBuffer);
					}
				else{
					char[] prepositionChar = Arrays.copyOfRange(conditionChar, 2, conditionWord[0].length());
					String prepositionWord = new String(prepositionChar);
					List<String> joined = new ArrayList<String>();
					joined.add(prepositionWord);
					joined.addAll(Arrays.asList(Arrays.copyOfRange(conditionWord, 1, conditionWord.length)));
					conditionBuffer += " " + String.join(" ", joined);
					return processTimeLocation(conditionList, 'L', conditionBuffer);
					}
				}
			else if (conditionChar[0] == 'T' && conditionChar[1] == ':'){
				if (Collections.frequency(Arrays.asList(timeMark), conditionWord[0].toLowerCase()) != 0){
					char[] processedChar = Arrays.copyOfRange(conditionChar, 2, conditionWord[0].length());
					String processedWord = new String(processedChar);
					List<String> joined = new ArrayList<String>();
					joined.add(processedWord.toLowerCase());
					joined.addAll(Arrays.asList(Arrays.copyOfRange(conditionWord, 1, conditionWord.length)));
					conditionBuffer += " " + String.join(" ", joined);
					return processTimeLocation(conditionList, 'T', conditionBuffer);
					}
				else if (Mark == 'T' && Collections.frequency(Arrays.asList(timeMark), conditionWord[0].toLowerCase()) == 0){
					char[] prepositionChar = Arrays.copyOfRange(conditionChar, 2, conditionWord[0].length());
					String processedWord = new String(prepositionChar);
					List<String> joined = new ArrayList<String>();
					joined.add(processedWord.toLowerCase());
					joined.addAll(Arrays.asList(Arrays.copyOfRange(conditionWord, 1, conditionWord.length)));
					conditionBuffer += " " + String.join(" ", joined);
					return processTimeLocation(conditionList, 'T', conditionBuffer);
					}
				else{
					conditionBuffer += " in " + String.join(" ", conditionWord);
					return processTimeLocation(conditionList, 'T', conditionBuffer);
					}
				}
			else{
				conditionBuffer += " " + condition;
				return processTimeLocation(conditionList, 'N', conditionBuffer);
			}
		}
	}

	private static ArrayList<String> processTuple(ArrayList<String> loadList){
		ArrayList<String> output = new ArrayList<String>();
		output.add(loadList.get(0));
		loadList.remove(0);
		output.add(loadList.get(0));
		loadList.remove(0);
		String objectBuffer = loadList.get(0);
		loadList.remove(0);
		char[] objectChar = objectBuffer.toCharArray();
		char Mark = 'N';
		if (objectChar[0] == 'T' && objectChar[1] == ':'){
			Mark = 'T';
			char[] temp = Arrays.copyOfRange(objectChar, 2, objectChar.length);
			objectBuffer = new String(temp);
			}
		if (objectChar[0] == 'L' && objectChar[1] == ':'){
			Mark = 'L';
			char[] temp = Arrays.copyOfRange(objectChar, 2, objectChar.length);
			objectBuffer = new String(temp);
			}
		String condition = processTimeLocation(loadList, Mark, "");
		output.add(objectBuffer + condition);
		return output;
	}
					 

}

File: framework/load.py
from sys import argv
script, filename, loadname = argv

locationMark = ['l:in', 'l:at', 'l:on', 'l:from']
timeMark = ['t:in', 't:at', 't:on', 't:from', 't:during', 't:between', 't:before' , 't:after', 't:while', 't:when']

def processTimeLocation(conditionList, Mark, conditionBuffer):
	if conditionList == []:
		return conditionBuffer
	else:
		condition = conditionList.pop(0)
		conditionChar = list(condition)
		conditionWord = condition.split()
		if conditionChar[0:2] == ['L',':']:
			if locationMark.count(conditionWord[0].lower()) == 0:
				prepositionChar = conditionChar[2: len(conditionWord[0])]
				prepositionWord = "".join(prepositionChar)
				conditionBuffer += " in " + " ".join([prepositionWord] + conditionWord[1:])
				return processTimeLocation(conditionList, 'L', conditionBuffer)
			else:
				prepositionChar = conditionChar[2: len(conditionWord[0])]
				prepositionWord = "".join(prepositionChar)
				conditionBuffer += " " + " ".join([prepositionWord] + conditionWord[1:])
				return processTimeLocation(conditionList, 'L', conditionBuffer)
		elif conditionChar[0:2] == ['T',':']:
			if not timeMark.count(conditionWord[0].lower()) == 0:
				processedChar = conditionChar[2: len(conditionWord[0])]
				processedWord = "".join(processedChar)
				conditionBuffer += " " + " ".join([processedWord.lower()] + conditionWord[1:])
				return processTimeLocation(conditionList, 'T', conditionBuffer)
			elif Mark == 'T' and timeMark.count(conditionWord[0].lower()) == 0:
				processedChar = conditionChar[2: len(conditionWord[0])]
				processedWord = "".join(processedChar)
				conditionBuffer += " " + " ".join([processedWord.lower()] + conditionWord[1:])
				return processTimeLocation(conditionList, 'T', conditionBuffer)
			else:
				conditionBuffer += " in " + " ".join(conditionWord)
				return processTimeLocation(conditionList, 'T', conditionBuffer)
		else:
			conditionBuffer += " " + condition
			return processTimeLocation(conditionList, 'None', conditionBuffer)

def processTuple(loadList):
	output = []
	output.append(loadList.pop(0))
	output.append(loadList.pop(0))
	objectBuffer = loadList.pop(0)
	objectChar = list(objectBuffer)
	Mark = 'None'
	if objectChar[0:2] == ['T',':']:
		Mark = 'T'
		temp = objectChar[2:]
		objectBuffer = "".join(temp)
	if objectChar[0:2] == ['L',':']:
		Mark = 'L'
		temp = objectChar[2:]
		objectBuffer = "".join(temp)
	condition = processTimeLocation(loadList, Mark, '')
	output.append(objectBuffer + condition)
	return output

outputFile = (open(filename)).read()
outputLines = outputFile.split('\n')
loadLines = []
sentenceCounter = 0
extractionCandidate = 0
sentenceExtraction = []

for line in outputLines:
	if line == '':
		if extractionCandidate == 1:
			for i in sentenceExtraction:
				loadLines.append(i)
		else:
			sentenceCounter -=1
		extractionCandidate = 0
		sentenceExtraction = []
	else:
		if not line[0].isdigit():
			sentenceCounter += 1
			loadLine = ["-----", ': '.join(["Line Number", str(sentenceCounter)]),"-----"]
			sentenceExtraction.append(loadLine)
		else:
			extractionCandidate = 1
			confidenceRate = float(line[0:3])
			if confidenceRate > 0.2:
				if not line[5] == '(':
					markPoint = line.index('(')
					keyWord = line[5:markPoint]
					startPoint = line.index(':') + 2
					content = '(' + keyWord + ')' + line[startPoint:-1]
				else:
					content = line[6:-1]
				loadLine = content.split('; ')
				if len(loadLine) >= 3:
					loadLine = processTuple(loadLine)
				sentenceExtraction.append(loadLine)

loadFile = open(loadname, 'w')
for i in loadLines:
	loadFile.write(i[0])
	loadFile.write("\n")
	loadFile.write(i[1])
	loadFile.write("\n")
	loadFile.write(i[2])
	loadFile.write("\n")
loadFile.close()
